pkg/prometheus/rules.go

```diff
@@ -122,13 +124,21 @@ func (c *Operator) createOrUpdateRuleConfigMaps(ctx context.Context, p *monitori
 		"namespace", p.Namespace,
 		"prometheus", p.Name,
 	)
-	for _, cm := range newConfigMaps {
-		_, err = cClient.Create(ctx, &cm, metav1.CreateOptions{})
+	for _, cm := range createOrUpdateConfigMaps {
+		err = k8sutil.CreateOrUpdateConfigMap(ctx, cClient, &cm)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create new ConfigMap '%v'", cm.Name)
+			return nil, errors.Wrapf(err, "failed to createOrUpdate ConfigMap '%v'", cm.Name)
 		}
 	}
 
+	// supple configmap number
```

Comment:
This comment has to be updated, it's important because it's the part of the code where we will return the extra empty configMaps that will be mounted optionally. This part of the code will not be obvious if the person reading the code does not know about #5085